backend/pkg/proto/service.go

```diff
@@ -473,3 +477,16 @@ func (s *Service) getFileDescriptorBySchemaID(schemaID int) (*desc.FileDescripto
 	desc, exists := s.fileDescriptorsBySchemaID[schemaID]
 	return desc, exists
 }
+
+type anyResolver struct {
+	mr *msgregistry.MessageRegistry
+}
+
+func (r *anyResolver) Resolve(typeURL string) (protoiface.MessageV1, error) {
+	mname := typeURL
+	if slash := strings.LastIndex(mname, "/"); slash >= 0 {
+		mname = mname[slash+1:]
+	}
```

Comment:
@chris-palmer-deltatre What's the reason for stripping the last slash? When would that occur, how is that an issue. Could you add a comment for that?

Reply:
@chris-palmer-deltatre Any chance that you remember this? Otherwise i'll try to reverse engineer it a bit

Reply:
Hi there apologies! Yes I do remember this bit. So basically I pulled this code from the internals of jhump/protoreflect's internal anyResolver. The reason its there is that type URLs often have the form "type.googleapis.com/myprotopackage.MyProtoType". However they will be indexed in the registry in the format "myprotopackage.MyProtoType". Therefore the contents before the last slash is removed so that type resolution can occur properly against the registry.

If I am honest, I am not sure about it in every case, but it seemed to work for the types I was using in the `Any` field for our purposes.

Reply:
So if you see the code here: https://github.dev/jhump/protoreflect/blob/master/dynamic/json.go#L1188

You will see that the internal `anyResolver` (which actually will invoke our custom `anyResolver` on line L1194 is strips the contents before the slash. However, when calling our custom `anyResolver` via `other.Resolve` on L1194, it passes the full typeURL which includes the contents before the slash. Therefore I copied the code from L1188 and put it in out custom `anyResolver` so I could replicate the same behavior.

Reply:
I think it makes sense because the internal `anyResolver` uses `mname` to query the registry anyway: https://github.com/jhump/protoreflect/blob/master/dynamic/json.go#L1203

We are also trying to query the registry in our custom `anyResolver`, so I guess it makes sense that I needed to replicate this behavior in order to get it working.

Reply:
Thanks @chris-palmer-deltatre , I tried to summarize that in a comment that I pushed into your branch for this PR